checks/raw/binary_artifact.go

```diff
@@ -40,6 +44,27 @@ func BinaryArtifacts(c clients.RepoClient) (checker.BinaryArtifactData, error) {
 		return checker.BinaryArtifactData{}, fmt.Errorf("%w", err)
 	}
 
+	// Indices of any gradle-wrapper.jar files
+	var gradleWrappers []int
+	if len(files) > 0 {
+		for i, f := range files {
+			if filepath.Base(f.Path) == "gradle-wrapper.jar" {
+				gradleWrappers = append(gradleWrappers, i)
+			}
+		}
+	}
+	if len(gradleWrappers) > 0 {
+		// Gradle wrapper JARs present, so check that they are validated
+		if ok, err := gradleWrapperValidationOK(c); ok && err == nil {
+			// It has been confirmed that latest commit has validated JARs!
+			// Remove Gradle wrapper JARs from files.
+			offset := 0
+			for _, ji := range gradleWrappers {
```

Comment:
This part of the code is a little hard to follow. Any simplification we can do?
```golang
binaries := []checker.File{}
err := fileparser.OnMatchingFileContentDo(c, fileparser.PathMatcher{
		Pattern:       "*",
		CaseSensitive: false,
	}, checkBinaryFileContent, &binaries)
...

files :=  []checker.File{}
for f := range binaries{
  if isGraddleJAR() {
   continue
  }

  // for other ecosystems
  if isMavenJAR() {
   continue
  }

  files = append(files, f)
}
```

Wdut?